### ISSUE TITLE
Guest Mode = syncing with mobile data

### ISSUE THREAD

**Issue by jangrewe** [syncthing/syncthing-android#458]
I just noticed that my Syncthing used 1.7 GB of traffic because it started syncing via my UMTS connection after i switched to Guest Mode (accidentaly).
It looks like it continued to sync even after i switched back to my main user, which i only noticed after my provider told me that i went over my limit.
Could you please make it ignore all settings except for the user who configured it? Thanks! =)

**Comment by AudriusButkevicius** [comment]
What all settings? I think the settings are local to the device, not the user.

**Comment by jangrewe** [comment]
Then it shouldn't start syncing (my whole storage with ~30GB of data) while ignoring that i explicitly set it to sync a) only on Wifi and b) only with power connected - but it synced while i was on vacation, using up all my roaming volume. That was a very expensive vacation...
So it should either respect all settings, including the restrictions, or not sync at all.

**Comment by Nutomic** [comment]
I'm not sure how this could happen. Android already seperates data from different apps (eg you can't access non-system apps or messages etc).

I just tried:
1. Set always run in background, only on power and wifi
2. Switch to guest account
3. Disable wifi
`top` showed that `libsyncthing.so` was stopped, and only the service was left running.

Having steps to reproduce this issue would help immensly. Android version and device name would also be interesting.

**Comment by AudriusButkevicius** [comment]
Yeah the point is that he started syncthing as a guest, which means Guest got it's own Java settings, though probably was still using same home directory for syncthing binary.

**Comment by Nutomic** [comment]
You can't even access another account's apps on a guest account (just tried on Android 5.1). So he would have to install the app manually for that to happen.

**Comment by AudriusButkevicius** [comment]
What if syncthing restarts, does it suddenly get thrown into a different sandbox? Can you not share apps across users?

**Comment by jangrewe** [comment]
This happened on a OnePlus One with CyanogenOS 12.0

Please take note that it didn't run as the Guest user - i only switched to Guest by accident, then switched back to my regular user.
Maybe Syncthing started once as Guest, overwrote some settings (initialized them?), and the when i switched back it used those reset settings?

**Comment by Nutomic** [comment]
If you switch to the guest account, you'll notice there isn't even a Syncthing item in the launcher, you have to install it manually. So it really has to be the background service from your main account running.

Not sure about different sandboxes, but that seems unlikely as it would break user seperation (and security).

Really, steps to reproduce would be the most helpful now, instead of guessing :p

**Comment by emdioh** [comment]
I just had a similar problem: I just noticed that my monthly mobile data quota is almost over and it looks like syncthing used most of it.
When I opened syncthing to check the settings it complained that it was disabled like it always does if it's set to wifi only and indeed it was set to sync on wifi only.
So: it wasn't supposed to use mobile data and it claimed that it wasn't even going to run. 
Maybe there is a bug somewhere and it ignores the setting from time to time?
I do a good amount of international roaming (mobile data is set to disabled when roaming) could it somehow be related?
